lms/djangoapps/verify_student/services.py

```diff
@@ -56,3 +63,26 @@ def start_verification(self, course_id, related_assessment, item_id):
             )
         )
         return re_verification_link
+
+    def get_attempts(self, user_id, course_id, related_assessment, location_id):
+        """
+        Get re-verification attempts against a user for a given 'checkpoint'
+        and 'course_id'.
+
+        Args:
+            user_id(str): User Id string
+            course_id(str): A string of course id
+            related_assessment(str): Verification checkpoint name
+            location_id(str): Reverification XBlock's location in courseware
+
+        Returns:
+            Number of re-verification attempts of a user
+        """
+        course_key = CourseKey.from_string(course_id)
+        return VerificationStatus.objects.filter(
+            user_id=user_id,
+            checkpoint__course_id=course_key,
+            checkpoint__checkpoint_name=related_assessment,
+            checkpoint__location_id=location_id,
+            status=VerificationStatus.VERIFICATION_STATUS_CHOICES.submitted
+        ).count()
```

Comment:
It might make sense to move this into a `classmethod` of `VerificationStatus`.  @aamir-khan wrote a very similar query in his PR for checking this when sending the email.